scripts/stubtest/allowlist.txt

```diff
@@ -551,8 +551,9 @@ django.contrib.gis.forms.BaseModelFormSet.save_m2m
 # Dynamically generated in https://github.com/django/django/blob/0ee06c04e0256094270db3ffe8b5dafa6a8457a3/django/core/mail/backends/locmem.py#L24
 django.core.mail.outbox
 
-# We use a trick using a descriptor class to represent an attribute with different type at the class and instance level.
-# Here to narrow `str | Pattern[str]` at the class level to `Pattern[str]` at the instance level.
+# The type system does not allow to represent an attribute with different type at the class level (`str | Pattern[str]`)
+# and instance (`Pattern[str]`) level. But in order to have more precise type at the instance level, we restrict the types
```

Comment:
> But in order to have more precise type at the instance level, we restrict the types

Sorry, we can't do that either. We can allow something that does not work to happen (false negative), but this will block valid useges (false positive). That's something we try to avoid, unless we are 100% sure that it is worth it.

Reply:
Then I guess we cannot do more currently. At least I did not find a way. I tried another approach using `__new__` but could not get it to work properly for both case either.

I'll close the PR, we might be able to revisit once the type system allows to represent this.

Thanks for the review nonetheless!

Reply:
Maybe #2615 will find a way